buildSrc/src/main/java/gradlebuild/NcursesRuntimePlugin.java

```diff
@@ -0,0 +1,109 @@
+package gradlebuild;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import dev.nokee.platform.base.BuildVariant;
+import dev.nokee.platform.base.Variant;
+import dev.nokee.platform.jni.JavaNativeInterfaceLibrary;
+import dev.nokee.runtime.nativebase.TargetMachineFactory;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.reflect.TypeOf;
+import org.gradle.api.specs.Spec;
+import org.gradle.nativeplatform.platform.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+
+import java.io.File;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import static gradlebuild.JavaNativeInterfaceLibraryUtils.library;
+import static gradlebuild.NcursesVersion.NCURSES_5;
+import static gradlebuild.NcursesVersion.NCURSES_6;
+import static java.util.Arrays.stream;
+
+@SuppressWarnings("UnstableApiUsage")
+public class NcursesRuntimePlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        project.getExtensions().add(new TypeOf<Provider<Set<NcursesVersion>>>() {}, "availableNcursesVersions", availableNcursesVersions(project));
+
+        project.getPluginManager().withPlugin("dev.nokee.jni-library", appliedPlugin -> {
+            library(project, library -> {
+                // Register ncurses dimension
+                SetProperty<NcursesVersion> ncurses = registerNcursesDimension(library);
+
+                // Defaults to all ncurses versions
+                ncurses.convention(ImmutableSet.copyOf(NcursesVersion.values()));
+
+                // Register dimension as library extensions
+                ((ExtensionAware) library).getExtensions().add("targetNcurses", ncurses);
+
+                library.getVariants().configureEach(ncursesVariant(), variant -> {
+                    variant.getSharedLibrary().getLinkTask().configure(task -> {
+                        task.getLinkerArgs().addAll(project.provider(ofNcursesLibraryFlags(library.getMachines(), variant.getBuildVariant())));
+                    });
+                });
+            });
+        });
+    }
+
+    private static Spec<Variant> ncursesVariant() {
+        return variant -> stream(NcursesVersion.values())
+            .anyMatch(it -> variant.getBuildVariant().hasAxisOf(it));
+    }
+
+    private static Callable<Iterable<String>> ofNcursesLibraryFlags(TargetMachineFactory machines, BuildVariant buildVariant) {
+        return () -> {
+            if (buildVariant.hasAxisOf(machines.getLinux().getOperatingSystemFamily()) && !buildVariant.hasAxisOf(NCURSES_5)) {
+                return ImmutableList.of("-lncursesw");
+            } else {
+                return ImmutableList.of("-lcurses");
+            }
+        };
```

Comment:
🔍 In theory, `curses`/`ncurses` library should be a normal dependency (added via `dependencies` DSL block). The main reason is allowing Nokee to use the dependency information to determine buildability among other things. This work will be done in the future with a deprecation cycle.